test/e2e/reporter.go

```diff
@@ -0,0 +1,258 @@
+	"github.com/openshift-kni/k8sreporter"
+	aprv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/adminpolicybasedroute/v1"
+	egressfwv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressfirewall/v1"
+	egressipv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressip/v1"
+	egressqosv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressqos/v1"
+	egressservicev1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressservice/v1"
+	"github.com/pkg/errors"
+	metallbv1beta1 "go.universe.tf/metallb/api/v1beta1"
+	metallbv1beta2 "go.universe.tf/metallb/api/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2ekubectl "k8s.io/kubernetes/test/e2e/framework/kubectl"
+	"k8s.io/utils/strings/slices"
+)
+
+const frrContainerName = "frr"
+
+func InitReporter(kubeconfig, path string, namespaces []string) *k8sreporter.KubernetesReporter {
+	// When using custom crds, we need to add them to the scheme
+	addToScheme := func(s *runtime.Scheme) error {
+		err := aprv1.AddToScheme(s)
+		if err != nil {
+			return err
+		}
+		err = egressipv1.AddToScheme(s)
+		if err != nil {
+			return err
+		}
+		err = egressfwv1.AddToScheme(s)
+		if err != nil {
+			return err
+		}
+		err = egressqosv1.AddToScheme(s)
+		if err != nil {
+			return err
+		}
+		err = egressservicev1.AddToScheme(s)
+		if err != nil {
+			return err
+		}
+		err = metallbv1beta1.AddToScheme(s)
+		if err != nil {
+			return err
+		}
+		err = metallbv1beta2.AddToScheme(s)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	// The namespaces we want to dump resources for (including pods and pod logs)
+	dumpNamespace := func(ns string) bool {
+		return slices.Contains(namespaces, ns)
+	}
+
+	// The list of CRDs we want to dump
+	crds := []k8sreporter.CRData{
+		{Cr: &aprv1.AdminPolicyBasedExternalRouteList{}},
+		{Cr: &egressipv1.EgressIPList{}},
+		{Cr: &egressfwv1.EgressFirewallList{}},
+		{Cr: &egressqosv1.EgressQoSList{}},
+		{Cr: &egressservicev1.EgressServiceList{}},
+		{Cr: &metallbv1beta1.IPAddressPoolList{}},
+		{Cr: &metallbv1beta1.AddressPoolList{}},
+		{Cr: &metallbv1beta2.BGPPeerList{}},
+		{Cr: &metallbv1beta1.L2AdvertisementList{}},
+		{Cr: &metallbv1beta1.BGPAdvertisementList{}},
+		{Cr: &metallbv1beta1.BFDProfileList{}},
+		{Cr: &metallbv1beta1.CommunityList{}},
+		{Cr: &corev1.ServiceList{}},
+	}
+
+	reporter, err := k8sreporter.New(kubeconfig, addToScheme, dumpNamespace, path, crds...)
+	if err != nil {
+		log.Fatalf("Failed to initialize the reporter %s", err)
+	}
+	return reporter
+}
+
+func DumpInfo(reporter *k8sreporter.KubernetesReporter, testName string) {
+	testNameNoSpaces := strings.ReplaceAll(ginkgo.CurrentSpecReport().LeafNodeText, " ", "-")
+	reporter.Dump(10*time.Minute, testNameNoSpaces)
+}
+
+func dumpBGPInfo(basePath, testName string, f *framework.Framework) {
```

Comment:
hmm I don't see the logic of exporting vs un-exporting methods.

Would it be reasonable to un-export the methods that are used only from this file, and export the others?

Reply:
yes, it doesn't need to be a public method for now, made it as private method with relevant comment.

Reply:
oh! yes, done.